lib/rbnacl/secret_boxes/xchacha20poly1305.rb

```diff
@@ -0,0 +1,141 @@
+# encoding: binary
+# frozen_string_literal: true
+
+module RbNaCl
+  module SecretBoxes
+    # The XChaCha20Poly1305 class boxes and unboxes messages using the XChaCha20Poly1305 algorithm
```

Comment:
It would probably be good to note there are two different flavors of algorithm calling themselves `XChaCha20Poly1305`

- `crypto_secretbox` / `djb`-style: this is effectively the `crypto_secretbox` function but with `XChaCha20` swapped for `XSalsa20`, and what is being implemented here
- [`draft-irtf-cfrg-xchacha`](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-xchacha-03): uses the IETF flavor of ChaCha20